File: app/src/main/java/com/cn29/aac/ui/shopping/ArtistFragmentModule.java
package com.cn29.aac.ui.shopping;

import com.cn29.aac.R;
import com.cn29.aac.databinding.FragmentArtistBinding;
import com.cn29.aac.ui.shopping.vm.ArtistFragmentViewModel;
import com.cn29.aac.ui.shopping.vm.ArtistFragmentViewModelFactory;
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;
import dagger.Module;
import dagger.Provides;

/**
 * Created by Charles Ng on 20/10/2017.
 */

@Module
public class ArtistFragmentModule {


  @Provides
  ArtistFragmentViewModel getArtistViewModel(ArtistFragmentViewModelFactory factory,
      ArtistFragment artistFragment) {
    return ViewModelProviders.of(artistFragment.getActivity(), factory)
        .get(ArtistFragmentViewModel.class);
  }

  @Provides
  ShoppingActivityViewModel getShoppingKartActivityViewModel(ArtistFragment artistFragment) {
    return ViewModelProviders.of(artistFragment.getActivity())
        .get(ShoppingActivityViewModel.class);
  }


  @Provides
  FragmentArtistBinding provideBinding(ArtistFragment artistFragment) {
    return DataBindingUtil
        .inflate(artistFragment.getActivity().getLayoutInflater(), R.layout.fragment_artist, null,
            false);
  }
}

File: app/src/main/java/com/cn29/aac/ui/shopping/ArtistFragmentModule.kt
package com.cn29.aac.ui.shopping

import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.cn29.aac.R
import com.cn29.aac.databinding.FragmentArtistBinding
import com.cn29.aac.ui.shopping.vm.ArtistFragmentViewModel
import com.cn29.aac.ui.shopping.vm.ArtistFragmentViewModelFactory
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel
import dagger.Module
import dagger.Provides


@Module
class ArtistFragmentModule {
    @Provides
    fun getArtistViewModel(factory: ArtistFragmentViewModelFactory,
                           artistFragment: ArtistFragment): ArtistFragmentViewModel {
        return ViewModelProvider(artistFragment.requireActivity(), factory)
                .get(ArtistFragmentViewModel::class.java)
    }

    @Provides
    fun getShoppingKartActivityViewModel(artistFragment: ArtistFragment): ShoppingActivityViewModel {
        return ViewModelProvider(artistFragment.requireActivity())
                .get(ShoppingActivityViewModel::class.java)
    }

    @Provides
    fun provideBinding(artistFragment: ArtistFragment): FragmentArtistBinding {
        return DataBindingUtil
                .inflate(artistFragment.requireActivity().layoutInflater,
                         R.layout.fragment_artist,
                         null,
                         false)
    }
}